skcosmo/decomposition/_kernel_pcovr.py

```diff
@@ -200,31 +214,54 @@ def __init__(
 
         self.mixing = mixing
         self.n_components = n_components
-        self.alpha = alpha
 
         self.svd_solver = svd_solver
         self.tol = tol
         self.iterated_power = iterated_power
         self.random_state = random_state
         self.center = center
 
-        self.kernel = kernel
-        self.kernel_params = kernel_params
-        self.gamma = gamma
-        self.degree = degree
-        self.coef0 = coef0
+        if regressor is None:
+            regressor = KernelRidge(
+                alpha=1e-6,
+                kernel="linear",
+                gamma=None,
+                degree=3,
+                coef0=1,
+                kernel_params=None,
+            )
+
+        self.regressor = regressor
+
         self.n_jobs = n_jobs
         self.n_samples_ = None
 
         self.fit_inverse_transform = fit_inverse_transform
 
     def _get_kernel(self, X, Y=None):
-        if callable(self.kernel):
-            params = self.kernel_params or {}
+
+        try:
+            check_is_fitted(self, "regressor_")
+            regressor_params = self.regressor_.get_params()
+
+        except NotFittedError:
+            regressor_params = self.regressor.get_params()
```

Comment:
I don't understand why this is here

Reply:
So `self.regressor` is mutable. If we init `KernelPCovR` with a regressor, and subsequently change the parameters of the regressor _outside_ of the `KernelPCovR` instance, it's `self.regressor` will also be updated. This is here to ensure that once the regressor of `KernelPCovR` has been fitted, the kernel parameters used to build the kernels won't change. Since `_get_kernel` is required in `predict` and `transform`, this means that here we have to use the fitted regressor parameters if they exist to ensure that the test kernel is using the same parameters as the train kernel.

Reply:
This will be obsolete if we re-add individual kernel parameters to the `__init__`